sonar-dotnet-shared-library/src/test/java/org/sonar/plugins/dotnet/tests/OpenCoverReportParserTest.java

```diff
@@ -129,16 +130,56 @@ public void branchCoverage() throws Exception {
         Assertions.entry(18, 1),
         Assertions.entry(25, 1));
 
-    assertThat(coverage.getBranchCoverage(filePath).size()).isEqualTo(5);
+    assertThat(coverage.getBranchCoverage(filePath))
+      .hasSize(5)
+      .contains(
+        new BranchCoverage(9, 4 , 2),
+        new BranchCoverage(12, 2 , 1),
```

Comment:
For the record,line 12 should be
```
            _ = y == 0 || z == 0;
```
and given the tests is invoked with `x = 0, y = 0, z = 0`, then the boolean short-circuit means that `z==0` is not executed at all -> CORRECT

Reply:
same for line 13

Reply:
same idea for line 14

Reply:
In this case the decompiled code is `int num1 = y == 0 ? 1 : (z == 0 ? 1 : 0);`
but the branches seem to be:
- y == 0 || z == 0 -> 1
- y != 0 && z != 0 -> 0

Fun fact: I would expect this, in the initial form (_ = y == 0 || z == 0;) to be a boolean not an int.